src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private static final Object[] DEFAULT_EMPTY_DATA_ARRAY = {};
+    private static final String INVALID_INDEX_ERROR_MESSAGE = 
+            "Index should be bigger then 0 and less then ArrayList size";
+
+    private Object[] dataArray;
+    private int size;
+
+    ArrayList() {
+        dataArray = DEFAULT_EMPTY_DATA_ARRAY;
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        dataArray[size] = value;
+        ++size;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        validateIndex(index);
+        growIfArrayFull();
+        System.arraycopy(dataArray, index, dataArray, index + 1, size - index);
+        dataArray[index] = value;
+        ++size;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
Don't create repeating code. You can utilize `add(T value, int index)` method inside the `addAll` method to avoid code duplication and handle the growth of the array there.